src/System.Text.Json/src/System/Text/Json/JsonHelpers.Date.cs

```diff
@@ -226,14 +226,13 @@ private static bool TryParseDateTimeOffset(ReadOnlySpan<byte> source, out DateTi
             // Parse fraction
             {
                 int numDigitsRead = 0;
+                int fractionEnd = Math.Min(sourceIndex + JsonConstants.DateTimeParseNumFractionDigits, source.Length);
 
-                while (sourceIndex < source.Length && IsDigit(curByte = source[sourceIndex]))
+                while (sourceIndex < fractionEnd && IsDigit(curByte = source[sourceIndex]))
                 {
-                    int prevFractionTimesTen = fraction * 10;
-
-                    if ((prevFractionTimesTen + (int)(curByte - (uint)'0') <= JsonConstants.MaxDateTimeFraction) && (numDigitsRead < JsonConstants.DateTimeNumFractionDigits))
+                    if (numDigitsRead < JsonConstants.DateTimeNumFractionDigits)
                     {
-                        fraction = prevFractionTimesTen + (int)(curByte - (uint)'0');
+                        fraction = (fraction * 10) + (int)(curByte - (uint)'0');
```

Comment:
Is this change necessary? Would constructing the DateTime with higher precision (beyond 7 digits) change the returned DateTime in anyway?

Reply:
Also, do we need to modify the debug assert here: https://github.com/dotnet/corefx/pull/36347/files#diff-e2e1b5ba65b1a8b77a086eb93e77c2d1R518?

Reply:
If we don't cap to 7 digits here, then modify the debug assert, we will diverge from the `Utf8Parser` helpers that we are currently trying to state consistent with (and eventually de-duplicate and rely on: https://github.com/dotnet/corefx/issues/35953).
Also, if we don't enforce a cap, we could overflow at 10 digits.